src/module.c

```diff
@@ -251,6 +250,7 @@ typedef struct RedisModuleBlockedClient {
     void *privdata;     /* Module private data that may be used by the reply
                            or timeout callback. It is set via the
                            RedisModule_UnblockClient() API. */
+    client *thread_safe_ctx_client;/* Fake client used for thread safe context.*/
```

Comment:
```suggestion
    client *thread_safe_ctx_client; /* Fake client to be used for thread safe context so that no lock is required. */
```

Reply:
Looking at this struct now, I wonder if we can just use the `reply_client`? Not sure though ...

Reply:
@MeirShpilraien I'm not sure either. It looks okay to use `reply_client` if user doesn't do something surprising. It works for my case and tests pass as well but I'm not sure so I didn't reuse it here. I'd love to use it as it means one less `client` struct to clear, leads to better performance.  

I think question is, for a thread safe context, can someone call functions other than `RM_reply**` functions? If no, we can use `reply_client` here I believe.

Reply:
i don't know enough about this.. so i'm paranoid.

Reply:
I think not, it's deliberately a different client:
https://github.com/redis/redis/blob/266d95066da8082ab569207765bea674fa297404/src/module.c#L6625-L6628